src/System.Text.Json/tests/Serialization/JsonElementTests.cs

```diff
@@ -145,5 +146,32 @@ public void Verify()
                 Assert.Equal("Hello", Array[3].ToString());
             }
         }
+
+        [Theory,
+            InlineData(5),
+            InlineData(10),
+            InlineData(20),
+            InlineData(1024)]
+        public void ReadJsonElementFromStream(int defaultBufferSize)
+        {
+            // Streams need to read ahead when they hit objects or arrays that are assigned to JsonElement or object.
+
+            byte[] data = Encoding.UTF8.GetBytes(@"{""Data"":[1,true,{""City"":""MyCity""},null,""foo""]}");
```

Comment:
How's the code coverage here?

Should we add various other edge cases like `{}`,`[]`, `{[]}`, empty data, high depth, just a simple array/object (no nesting), etc.

Reply:
We're reasonably covered by all of the shared class test in SpanTests.

Reply:
Isn't this special? There are different cases where reading from a stream where a partial token or partial object is available in the first chunk, and where `TrySkip` returns false, for each "token type". I didn't see a lot of "partial input" test cases.

Reply:
We only TrySkip for StartObject and StartArray.